internal/cmd/satellites.go

```diff
@@ -0,0 +1,217 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"cdr.dev/coder-cli/internal/x/xcobra"
+
+	"github.com/spf13/cobra"
+	"golang.org/x/xerrors"
+
+	"cdr.dev/coder-cli/coder-sdk"
+	"cdr.dev/coder-cli/pkg/clog"
+	"cdr.dev/coder-cli/pkg/tablewriter"
+)
+
+const (
+	satelliteKeyPath = "/api/private/satellites/key"
+)
+
+type satelliteKeyResponse struct {
+	Key         string `json:"key"`
+	Fingerprint string `json:"fingerprint"`
+}
+
+func satellitesCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "satellites",
+		Short: "Interact with Coder satellite deployments",
+		Long:  "Perform operations on the Coder satellites for the platform.",
+	}
+
+	cmd.AddCommand(
+		createSatelliteCmd(),
+		listSatellitesCmd(),
+		deleteSatelliteCmd(),
+	)
+	return cmd
+}
+
+func createSatelliteCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create [name] [satellite_access_url]",
+		Args:  xcobra.ExactArgs(2),
+		Short: "create a new satellite.",
+		Long:  "Create a new Coder satellite.",
+		Example: `# create a new satellite
+
+coder satellites create eu-west https://eu-west-coder.com`,
```

Comment:
```suggestion
coder satellites create eu-west https://eu-west.coder.com`,
```

Reply:
Why? Part of the reason I picked that was to illustrate it could be a completely separate domain if desired.

Reply:
Well we don't own `eu-west.coder.com` so we'd be displaying a domain name that anyone could buy.

Reply:
Ah fair